packages/api-format/src/codec/sample.json

```diff
@@ -0,0 +1,2199 @@
+          {
+            "name": "PublicPropCount",
+            "modifier": 1,
+            "type": "PropIndex",
+            "documentation": [
+              " The number of (public) proposals that have been made so far."
+            ]
+          },
+          {
+            "name": "PublicProps",
+            "modifier": 1,
+            "type": "Vec<(PropIndex, T::Proposal, T::AccountId)>",
+            "documentation": [
+              " The public proposals. Unsorted."
+            ]
+          },
+          {
+            "name": "DepositOf",
+            "modifier": 0,
+            "type": {
+              "key": "PropIndex",
+              "value": "(T::Balance, Vec<T::AccountId>)"
+            },
+            "documentation": [
+              " Those who have locked a deposit."
+            ]
+          },
+          {
+            "name": "LaunchPeriod",
+            "modifier": 2,
+            "type": "T::BlockNumber",
+            "documentation": [
+              " How often (in blocks) new public referenda are launched."
+            ]
+          },
+          {
+            "name": "MinimumDeposit",
+            "modifier": 2,
+            "type": "T::Balance",
+            "documentation": [
+              " The minimum amount to be used as a deposit for a public referendum proposal."
+            ]
+          },
+          {
+            "name": "VotingPeriod",
+            "modifier": 2,
+            "type": "T::BlockNumber",
+            "documentation": [
+              " How often (in blocks) to check for new votes."
+            ]
+          },
+          {
+            "name": "ReferendumCount",
+            "modifier": 2,
+            "type": "ReferendumIndex",
+            "documentation": [
+              " The next free referendum index, aka the number of referendums started so far."
+            ]
+          },
+          {
+            "name": "NextTally",
+            "modifier": 2,
+            "type": "ReferendumIndex",
+            "documentation": [
+              " The next referendum index that should be tallied."
+            ]
+          },
+          {
+            "name": "ReferendumInfoOf",
+            "modifier": 0,
+            "type": {
+              "key": "ReferendumIndex",
+              "value": "(T::BlockNumber, T::Proposal, VoteThreshold)"
+            },
+            "documentation": [
+              " Information concerning any given referendum."
+            ]
+          },
+          {
+            "name": "VotersFor",
+            "modifier": 1,
+            "type": {
+              "key": "ReferendumIndex",
+              "value": "Vec<T::AccountId>"
+            },
+            "documentation": [
+              " Get the voters for the current proposal."
+            ]
+          },
+          {
+            "name": "VoteOf",
+            "modifier": 0,
+            "type": {
+              "key": "(ReferendumIndex, T::AccountId)",
+              "value": "bool"
+            },
```

Comment:
Where these `type: { key: ..., value: ... }` are found, it is actually a `Map`. so key becomes the param that is passed, and value is the result type, i.e. type.

Reply:
Oooh, and I picked an interesting one - the map key is actually a struct as well (it is ok, when we call it gets encoded, so ends up just being a u8a)